### PR TITLE
[WIP] Tests/NTK preparation/losses

### PR DESCRIPTION
**Pull request by marcelodallaqua**
Each commit was defined with a specific purpose. Please try to see the commits individually.

### CODE REVIEW

**Comment by levimcclenny** [comment]
Can we fix the instances where "weight" is misspelled - such as `initialize_weigths_loss()`, etc?

Still reviewing - can you add the [WIP] tag to the beginning of the PR as well?

Looking good! Thanks!

**Comment by marcelodallaqua** [comment]
Sorry, every time there is this " ght" on the English words, I get confused.

I not sure how to fix this now. I am thinking about making a new pull
request. What do you think?

And, what "WIP" means?

Marcelo

File: tensordiffeq/models.py [hunk 1]
```diff
@@ -13,18 +13,23 @@ class CollocationSolverND:
...
             layer_sizes: A list of layer sizes, can be overwritten via resetting u_model to a keras model
             f_model: PDE definition
             domain: a Domain object containing the information on the domain of the system
             bcs: a list of ICs/BCs for the problem
-            isAdaptive: Boolean value determining whether to implement self-adaptive solving
+            Adaptive_type: string with the adaptive method
+                                1 - Self-adaptive (https://arxiv.org/pdf/2009.04544.pdf),
+                                2 - Self-adaptive_loss with weights for the entire loss function,
+                                3 - NTK
+                                4 - None (no adaptive method)
```
Comment (levimcclenny) [inline tensordiffeq/models.py:30]: Should we make this into a dictionary so users select an int value such as '4' and it selects the scheme? that way we don't have to worry about spelling, etc.

**Comment by levimcclenny** [comment]
I added the WIP tag on the PR (work-in-progress) just meaning that we're going to go through some iterations before its merged. 

The weight misspellings appear to be fixed - I think we can remove the plotting_V2 and make that a separate PR. 

What is the main addition of this set of commits? Moving forward I would suggest just adding one feature at a time and opening a PR on just that, so its less at once. The we can have a commit that just says "NTK weighting" and we know thats all the NTK weighting stuff, etc.

**Comment by marcelodallaqua** [comment]
Please cancel the plotting_V2. This is my personal code and I was not intending to upload to this PR

**Comment by levimcclenny** [comment]
Alright the plotting was removed. Any comment with making the weighting scheme a dictionary that users choose from using an int?

**Comment by marcelodallaqua** [comment]
Perfect.

The weighing scheme with int is a great idea. I'll implement it
Monday morning.

Weekend are hard to work with children :(

Have a nice Sunday
Marcelo

**Comment by levimcclenny** [comment]
Closing to merge features in smaller PRs

**Pull request closed without merging**